### PR TITLE
Add H3 to `examples/hpc-enterprise-slurm`

### PR DESCRIPTION
**Pull request by mr0re1**

### CODE REVIEW

File: examples/hpc-enterprise-slurm.yaml [hunk 1]
```diff
@@ -255,10 +255,41 @@ deployment_groups:
...
+      node_count_dynamic_max: 16
+      machine_type: h3-standard-88
+      instance_image:
+        family: $(vars.family)
+        project: $(vars.project)
+      service_account:
+        email: $(compute_sa.service_account_email)
+        scopes:
+        - https://www.googleapis.com/auth/cloud-platform
+      disk_type: pd-balanced # H3 does not support pd-ssd and pd-standard
```
Comment (cboneti) [inline examples/hpc-enterprise-slurm.yaml:270]: what other type of PD is supported?

File: examples/hpc-enterprise-slurm.yaml [hunk 2]
```diff
@@ -255,10 +255,41 @@ deployment_groups:
...
+      disk_size_gb: 100
+      node_conf: # one has to specify number of CPUs and sockets for H3 machine type
+        CPUs: 88
+        CoresPerSocket: 44
+        Sockets: 2
+        ThreadsPerCore: 1
```
Comment (cboneti) [inline examples/hpc-enterprise-slurm.yaml:276]: Is this going to get fixed by SchedMD?

File: examples/hpc-enterprise-slurm.yaml [hunk 3]
```diff
@@ -255,10 +255,41 @@ deployment_groups:
...
+
+  # use `-p h3` to submit jobs to this partition:
+  # ex: `srun -p h3  -N 1 hostname`
+  - id: h3_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v5-partition
+    use: [h3_node_group, network1, homefs, projectsfs, scratchfs]
+    settings:
+      partition_name: h3
+      partition_conf: # one has to specify mem per CPU for H3 machine type
+        DefMemPerCPU: 3971
```
Comment (cboneti) [inline examples/hpc-enterprise-slurm.yaml:286]: Is this planned to get fixed?

**Review by cboneti — COMMENTED** [review]
Just a few questions

File: examples/hpc-enterprise-slurm.yaml (hunk 1 above)
Comment (mr0re1) [reply examples/hpc-enterprise-slurm.yaml:270]: `pd-balanced` only, see https://cloud.google.com/compute/docs/compute-optimized-machines#h3_disks

Comment (cboneti) [reply examples/hpc-enterprise-slurm.yaml:270]: please add the link to the blueprint.

File: examples/hpc-enterprise-slurm.yaml [hunk 4]
```diff
@@ -255,10 +255,41 @@ deployment_groups:
       # https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/develop/community/modules/compute/schedmd-slurm-gcp-v5-partition#compute-vm-zone-policies
       zones: $(vars.gpu_zones)
 
+  - id: h3_node_group
+    source: community/modules/compute/schedmd-slurm-gcp-v5-node-group
+    settings:
+      node_count_dynamic_max: 16
+      machine_type: h3-standard-88
```
Comment (cboneti) [inline examples/hpc-enterprise-slurm.yaml:262]: I would still explicitly add "bandwidth_tier: gvnic_enabled" and the link to where it talks about the network characteristics.

**Review by cboneti — APPROVED** [review]
Thanks!

**Pull request merged**